### PR TITLE
Interceptor

### PR DESCRIPTION
**Pull request by otvertka12335**
Added interceptors to project
Resolved conflicts with role-guard

### CODE REVIEW

**Comment by Kudesnitsa** [comment]
You need to delete the .idea folder from the root

File: frontend/src/app/layouts/admin-page/reports/report.component.html [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<div cdkDropList cdkDropListOrientation="horizontal" class="list" (cdkDropListDropped)="drop($event)">
```
Comment (inventorsoft-academy) [inline frontend/src/app/layouts/admin-page/reports/report.component.html]: Pull request name is **Interceptor**, so I assume to see the interceptor code :) Please, clean this PR to include only changes that are relevant to interceptor

File: frontend/src/app/services/auth.interceptor.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import {HttpEvent, HttpHandler, HttpInterceptor, HttpRequest} from "@angular/common/http";
```
Comment (inventorsoft-academy) [inline frontend/src/app/services/auth.interceptor.ts]: I don't see refresh token logic here based on token expiration. Please, add it

**Comment by inventorsoft-academy** [comment]
Most of the interceptor logic was rewritten, so it makes sense to create new pull request instead of updating this one

**Pull request closed without merging**